### PR TITLE
LPS-142896

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Related Issue: https://issues.liferay.com/browse/LPS-142896

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+				_createFieldValue(
+					"expirationDate", _getString(expirationDate)));
+		}
+
+		if (!hasLimitToOneSubmissionPerUser) {
+			fieldValuesJSONArray.put(
+				_createFieldValue(
+					"limitToOneSubmissionPerUser",
+					String.valueOf(expirationDate != null)));
+		}
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java:218]: @rodrigopaulino, can you please give more details why the value of `` is being set according the `expirationDate` actual value?

I apologize to you, but I didn't get the reason. :grin:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+					"neverExpire", String.valueOf(expirationDate == null)));
+		}
+
+		if (!hasShowPartialResultsToRespondents) {
+			fieldValuesJSONArray.put(
+				_createFieldValue(
+					"showPartialResultsToRespondents",
+					String.valueOf(expirationDate != null)));
+		}
+	}
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java:232]: @rodrigopaulino, can you please give more details why the value of `` is being set according the `expirationDate` actual value?

I apologize to you, but I didn't get the reason. :grin:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+	public DDMFormInstanceUpgradeProcess(JSONFactory jsonFactory) {
+		_jsonFactory = jsonFactory;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement selectPreparedStatement1 =
+				connection.prepareStatement(
+					"select expirationDate, formInstanceId, settings_ from " +
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java:46]: @rodrigopaulino, do you see any value we keep the column `expirationDate` in the database table after this upgrade is run?

From what I could understand, the expiration date info will be in this column and also in the column `settings_`, does it make sense?

Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java:46]: @natocesarrego, for now, I see. The expirationDate inside settings_ still doesn't contain the time information. Only after the following story we will deal with it. So we need the column to hold that data for now. We can erase it after we finish the Polls migration.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java (hunk 2 above)
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java:232]: @natocesarrego, if it contains expiration date, it means that it used to be a Poll, so we'll turn this setting on to show the partial results, as it was with the Polls product.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java (hunk 1 above)
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java:218]: @natocesarrego, if it contains expiration date, it means that it used to be a Poll, so we'll turn this setting on to show limit the number of submission to one, as it was with the Polls product.

**Comment by natocesarrego** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
I'm checking to see if this failure is related.

**Comment by rodrigopaulino** [comment]
@natocesarrego, I had to make some adjustments to this PR because it was failing tests. Apparently, since the `expirationDate` column is just created in the `PollsToDDMUpgradeProcess` and not added to the `service.xml`, it is possible to have a `DDMFormInstance` table without that column if the database is created after the inclusion of `PollsToDDMUpgradeProcess`. This scenario caused the new upgrade process to fail because it expects from the `DDMFormInstance` table to always have that column. The changes I've made was to take into account in the new upgrade process the possibility of not having that column. I chose to do that instead of adding the column to the service, because we will eventually delete it.
So could you please review this again?

Thanks in advance.

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by natocesarrego** [comment]
Good job, @rodrigopaulino! :slightly_smiling_face:

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**